src/components/ToolButton.tsx

```diff
@@ -41,20 +43,39 @@ const ToolButton: React.FC<ToolButtonProps> = ({
   onClick,
   icon,
   tooltipText,
+  toggle = false,
 }) => {
+  const [isToggled, setIsToggled] = useState(false);
+  const handleClick = (event: MouseEvent<HTMLButtonElement>) => {
+    if (toggle) {
+      setIsToggled(!isToggled);
+    }
+    onClick(event);
+  };
```

Comment:
Why should the toggled state change? Toggle buttons change between selected and unselected state, but they should not change the toggled state.